x/wasm/keeper/relay_test.go

```diff
@@ -329,73 +329,93 @@ func TestOnRecvPacket(t *testing.T) {
 		expPanic           bool
 		expEventTypes      []string
 	}{
-		"consume contract gas": {
+		"contract returns success ack": {
 			contractAddr:   example.Contract,
 			expContractGas: myContractGas,
-			contractResp: &wasmvmtypes.IBCReceiveResponse{
-				Acknowledgement: []byte("myAck"),
+			contractResp: &wasmvmtypes.IBCReceiveResult{
+				Ok: &wasmvmtypes.IBCReceiveResponse{Acknowledgement: []byte("myAck")},
 			},
 			expAck: []byte("myAck"),
 		},
-		"can return empty ack": {
+		"can return empty ack data": {
 			contractAddr:   example.Contract,
 			expContractGas: myContractGas,
-			contractResp:   &wasmvmtypes.IBCReceiveResponse{},
+			contractResp: &wasmvmtypes.IBCReceiveResult{
+				Ok: &wasmvmtypes.IBCReceiveResponse{},
+			},
 		},
-		"consume gas on error, ignore events + messages": {
+		"contract Err result converted to error Ack": {
+			contractAddr:   example.Contract,
+			expContractGas: myContractGas,
+			contractResp: &wasmvmtypes.IBCReceiveResult{
+				Err: "my-error",
+			},
+			expAck: []byte(`{"error":"my-error"}`), // without error msg redaction
+		},
+		"contract aborts tx with error": {
 			contractAddr:   example.Contract,
 			expContractGas: myContractGas,
-			contractResp: &wasmvmtypes.IBCReceiveResponse{
-				Acknowledgement: []byte("myAck"),
-				Messages:        []wasmvmtypes.SubMsg{{ReplyOn: wasmvmtypes.ReplyNever, Msg: wasmvmtypes.CosmosMsg{Bank: &wasmvmtypes.BankMsg{}}}},
-				Attributes:      []wasmvmtypes.EventAttribute{{Key: "Foo", Value: "Bar"}},
+			contractResp: &wasmvmtypes.IBCReceiveResult{
+				Ok: &wasmvmtypes.IBCReceiveResponse{
+					Acknowledgement: []byte("myAck"),
+					Messages:        []wasmvmtypes.SubMsg{{ReplyOn: wasmvmtypes.ReplyNever, Msg: wasmvmtypes.CosmosMsg{Bank: &wasmvmtypes.BankMsg{}}}},
```

Comment:
Why include contractResp here? If it errors, it shouldn't have a response

Reply:
Correct. Copy paste error. I will remove this